File: Algorithms/src/3577. Count the Number of Computer Unlocking Permutations/Code/solution.py
class Solution:
    def countPermutations(self, complexity: List[int]) -> int:
        MOD = 10**9 + 7
        n = len(complexity)
        
        # 1. Find global minimum and its count
        min_val = min(complexity)
        cnt_min = complexity.count(min_val)
        
        # 2. Check if index 0 has unique minimum
        if complexity[0] != min_val or cnt_min != 1:
            return 0
        
        # 3. Compute (n - 1)! % MOD
        ans = 1
        for i in range(2, n):
            ans = (ans * i) % MOD
        return ans

File: Algorithms/src/3577. Count the Number of Computer Unlocking Permutations/Code/solution.go
package main

func countPermutations(complexity []int) int {
	const MOD int64 = 1_000_000_007
	n := len(complexity)

	// 1. Find global minimum and its count
	minVal := complexity[0]
	cntMin := 0
	for _, x := range complexity {
		if x < minVal {
			minVal = x
			cntMin = 1
		} else if x == minVal {
			cntMin++
		}
	}

	// 2. Check if index 0 has unique minimum
	if complexity[0] != minVal || cntMin != 1 {
		return 0
	}

	// 3. Compute (n - 1)! % MOD
	ans := int64(1)
	for i := 2; i <= n-1; i++ {
		ans = (ans * int64(i)) % MOD
	}
	return int(ans)
}
